File: project1/backend/kitting-course-platform/src/main/java/com/example/knitting_course_platform/entity/LecturePattern.java
package com.example.knitting_course_platform.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "LECTURE_PATTERN")
@Getter
@NoArgsConstructor
public class LecturePattern {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "pattern_id")
    private Long patternId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "lecture_id", nullable = false)
    private Lecture lecture;

    @Column(name = "row_num", nullable = false)
    private Integer rowNum;

    @Column(name = "pattern_text", columnDefinition = "TEXT", nullable = false)
    private String patternText;
}

File: project1/front/lib/features/lecture/domain/entities/lecture_pattern.dart
import 'pattern_row.dart';

class LecturePattern implements PatternRow {
  @override final int patternId;
  final int lectureId;
  @override final int rowNum;
  @override final String patternText;

  const LecturePattern({
    required this.patternId,
    required this.lectureId,
    required this.rowNum,
    required this.patternText,
  });
}
